src/field/fieldgenerators.cxx

```diff
@@ -268,13 +269,148 @@ std::shared_ptr<FieldGenerator> FieldTanhHat::clone(const list<std::shared_ptr<F
   return std::shared_ptr<FieldGenerator>( new FieldTanhHat(xin, widthin, centerin, steepnessin));
 }
 
-BoutReal FieldTanhHat::generate(double x, double y, double z, double t) {
+BoutReal FieldTanhHat::generate(double x, double y, double z, double t, const DataIterator &i, Mesh *localmesh) {
   // The following are constants
-  BoutReal w = width    ->generate(0,0,0,0);
-  BoutReal c = center   ->generate(0,0,0,0);
-  BoutReal s = steepness->generate(0,0,0,0);
+  BoutReal w = width    ->generate(0,0,0,0,i,localmesh);
+  BoutReal c = center   ->generate(0,0,0,0,i,localmesh);
+  BoutReal s = steepness->generate(0,0,0,0,i,localmesh);
   return 0.5*(
-                 tanh( s*(X->generate(x,y,z,t) - (c - 0.5*w)) )
-               - tanh( s*(X->generate(x,y,z,t) - (c + 0.5*w)) )
+                 tanh( s*(X->generate(x,y,z,t,i,localmesh) - (c - 0.5*w)) )
+               - tanh( s*(X->generate(x,y,z,t,i,localmesh) - (c + 0.5*w)) )
              );
 }
+
+BoutReal FieldRealY::generate(double x, double y, double z, double t,
+                              const DataIterator &i, Mesh *localmesh) {
+  BoutReal *offset = doCache(localmesh);
+  int gny = localmesh->GlobalNy - mesh->ystart * 2;
+  int iy = y / 2 / PI * gny * 2 + 10.5;
```

Comment:
What's the origin of the 10.5 here?

Reply:
10: a sufficently large, even integer to be always positive. evenness to preserve staggering info
.5: so that the cast rounds appropriately.